pymove/tests/core/test_data_frame.py

```diff
@@ -0,0 +1,1102 @@
+    assert DIST_PREV_TO_NEXT not in move_df
+
+    move_df.generate_dist_features()
+    print(move_df)
+    assert_frame_equal(move_df, expected)
+
+
+def test_generate_time_features():
+    move_df = _default_move_df()
+
+    new_move_df = move_df.generate_time_features(inplace=False)
+    expected = DataFrame(data=[
+        [1, 39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), nan, 1.0, nan],
+        [1, 39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1.0, nan, nan],
+        [2, 39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), nan, 0.0, nan],
+        [2, 39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 0.0, nan, nan],
+    ], columns=['id', 'lat', 'lon', 'datetime', 'time_to_prev', 'time_to_next',
+                'time_prev_to_next'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(new_move_df, expected)
+    assert isinstance(new_move_df, PandasMoveDataFrame)
+    assert TIME_PREV_TO_NEXT not in move_df
+
+    move_df.generate_time_features()
+    print(move_df)
+    assert_frame_equal(move_df, expected)
+
+
+def test_generate_speed_features():
+    move_df = _default_move_df()
+
+    new_move_df = move_df.generate_speed_features(inplace=False)
+    expected = DataFrame(data=[
+        [0, 39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1, nan,
+         13.690153134343689, nan],
+        [1, 39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1,
+         13.690153134343689, nan, nan],
+        [2, 39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, nan, nan, nan],
+        [3, 39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, nan, nan, nan],
+    ], columns=['index', 'lat', 'lon', 'datetime', 'id', 'speed_to_prev', 'speed_to_next',
+                'speed_prev_to_next'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(new_move_df, expected)
+    assert isinstance(new_move_df, PandasMoveDataFrame)
+    assert SPEED_PREV_TO_NEXT not in move_df
+
+    move_df.generate_speed_features()
+    print(move_df)
+    assert_frame_equal(move_df, expected)
+
+
+def test_generate_move_and_stop_by_radius():
+    move_df = _default_move_df()
+
+    new_move_df = move_df.generate_move_and_stop_by_radius(inplace=False)
+    expected = DataFrame(data=[
+        [1, 39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), nan,
+         13.690153134343689, nan, 'nan'],
+        [1, 39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 13.690153134343689,
+         nan, nan, 'move'],
+        [2, 39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), nan, 0.0, nan,
+         'nan'],
+        [2, 39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 0.0, nan, nan,
+         'stop'],
+    ], columns=['id', 'lat', 'lon', 'datetime', 'dist_to_prev', 'dist_to_next',
+                'dist_prev_to_next', 'situation'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(new_move_df, expected)
+    assert isinstance(new_move_df, PandasMoveDataFrame)
+    assert SITUATION not in move_df
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574968)